pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -326,12 +326,22 @@ private SuccessResponse uploadSegment(FormDataMultiPart multiPart, boolean enabl
     }
   }
 
+  private FileUploadDownloadClient.UploadVersion getUploadVersion(String uploadVersionString) {
+    if (uploadVersionString != null) {
+      return FileUploadDownloadClient.UploadVersion.valueOf(uploadVersionString);
+    } else {
+      return FileUploadDownloadClient.UploadVersion.getDefaultUploadVersion();
+    }
+  }
+
   private String getZkDownloadURIForURIUpload(String currentSegmentLocationURI, SegmentMetadata segmentMetadata,
-      FileUploadPathProvider provider) throws URISyntaxException, UnsupportedEncodingException {
+      FileUploadPathProvider provider, boolean moveSegmentToFinalLocation) throws URISyntaxException, UnsupportedEncodingException {
     String zkDownloadUri;
     if (new URI(currentSegmentLocationURI).getScheme().equals(CommonConstants.Segment.LOCAL_SEGMENT_SCHEME)) {
       zkDownloadUri = ControllerConf.constructDownloadUrl(segmentMetadata.getTableName(), segmentMetadata.getName(),
           provider.getVip());
+    } else if (moveSegmentToFinalLocation) {
```

Comment:
shall this be if  (!moveSegmentToFinalLocation) ?

Reply:
good catch, fixed